android/app/src/main/java/app/organicmaps/sdk/sync/NextcloudSyncClient.java

```diff
@@ -0,0 +1,611 @@
+package app.organicmaps.sdk.sync;
+
+import android.net.Uri;
+import android.os.Handler;
+import android.os.HandlerThread;
+import android.util.Base64;
+import android.util.Xml;
+import androidx.annotation.Nullable;
+import app.organicmaps.sdk.util.FileUtils;
+import app.organicmaps.sdk.util.Utils;
+import app.organicmaps.sdk.util.log.Logger;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.URLDecoder;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+import okhttp3.MediaType;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+import okhttp3.ResponseBody;
+import org.xmlpull.v1.XmlPullParser;
+import org.xmlpull.v1.XmlPullParserException;
+
+public class NextcloudSyncClient extends SyncClient
```

Comment:
Same here. A huge class. Even GitHub displays it as collapsed in code browser by default.